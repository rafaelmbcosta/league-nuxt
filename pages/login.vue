<template>
  <div>
    <h1> LOGIN </h1>
    <v-card class="mt-5" :loading="loading">
      <v-card-text>
        <v-text-field v-model="email" :disabled="loading" label="EMAIL" />
        <v-text-field v-model="password" :disabled="loading" type="password" label="PASSWORD" />
      </v-card-text>
      <v-btn :disabled="loading" class="ma-5" color="primary" @click="login({email, password})">
        CONFIRMAR
      </v-btn>
    </v-card>
  </div>
</template>

<script>
import { mapActions, mapState } from 'vuex'

export default {
  data () {
    return {
      email: 'rafael@gmail.com',
      password: '123456'
    }
  },
  computed: {
    ...mapState('util', ['loading'])
  },
  methods: {
    ...mapActions('util', ['login'])
  }
}
</script>

<style>

</style>
