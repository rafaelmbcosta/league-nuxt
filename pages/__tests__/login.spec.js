import { shallowMount, createLocalVue } from "@vue/test-utils"
import Vuetify from "vuetify"
import login from "../login"

const localVue = createLocalVue()
const vuetify = new Vuetify()

describe("login", () => {
  let store

  const factory = () => shallowMount(login, {
    computed: {
      loading: false
    }
  }, localVue, vuetify, store)

  test("loads component", () => {
    const wrapper = factory()
    expect(wrapper).toMatchSnapshot()
  })

  test("calling login actions", () => {
    const wrapper = factory()
    const email = wrapper.find(".v-text-field")
    console.log(wrapper, "wrapper")
    console.log(email, "email")
    it.todo("fill in fields")
    it.todo("click the buttton")
    it.todo("expect action to be called")
  })
})
