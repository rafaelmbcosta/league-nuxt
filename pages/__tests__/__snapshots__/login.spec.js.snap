// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login loads component 1`] = `
<div>
  <h1> LOGIN </h1>
  <v-card-stub loaderheight="4" tag="div" class="mt-5">
    <v-card-text-stub tag="div">
      <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="EMAIL" loaderheight="2" clearicon="$clear" type="text"></v-text-field-stub>
      <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="PASSWORD" loaderheight="2" clearicon="$clear" type="password"></v-text-field-stub>
    </v-card-text-stub>
    <v-btn-stub color="primary" tag="button" activeclass="" type="button" class="ma-5">
      CONFIRMAR
    </v-btn-stub>
  </v-card-stub>
</div>
`;
